### PR TITLE
Tests and Docs for default EmbeddingModel methods.

### PR DESCRIPTION
**Pull request by crutcher**

### CODE REVIEW

File: langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java [hunk 1]
```diff
@@ -13,18 +13,34 @@
...
+     * @return the embedding.
+     */
     default Response<Embedding> embed(TextSegment textSegment) {
         Response<List<Embedding>> response = embedAll(singletonList(textSegment));
-        return Response.from(
-                response.content().get(0),
-                response.tokenUsage(),
-                response.finishReason()
-        );
+        assert response.content().size() == 1 :
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java]: Aren't assertions disabled by default? maybe better to use `ValidationUtils.ensureTrue(...)` or `ValidationUtils.ensureBetween(response.content().size(), 1, 1, "content")`

File: langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java [hunk 2]
```diff
@@ -13,18 +13,34 @@
...
+     *
+     * @param text the text to embed.
+     * @return the embedding.
+     */
     default Response<Embedding> embed(String text) {
         return embed(TextSegment.from(text));
     }
 
+    /**
+     * Embed a text segment.
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java]: Probably worth mentioning that `TextSegment.metadata` is not embedded, just the `TextSegment.text`.

File: langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java [hunk 3]
```diff
@@ -13,18 +13,34 @@
...
-                response.tokenUsage(),
-                response.finishReason()
-        );
+        assert response.content().size() == 1 :
+                String.format("Expected a single embedding, but got %d", response.content().size());
+        return Response.from(response.content().get(0), response.tokenUsage(), response.finishReason());
     }
 
+    /**
+     * Embeds a list of text segments.
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java]: Probably worth mentioning that `TextSegment.metadata` is not embedded, just the `TextSegment.text`.

**Review by ghost — CHANGES_REQUESTED** [review]
@crutcher thanks a lot! Please check a few minor comments

File: langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java (hunk 1 above)
Comment (crutcher) [reply langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java]: Done

File: langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java (hunk 3 above)
Comment (crutcher) [reply langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java]: Done

File: langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java (hunk 2 above)
Comment (crutcher) [reply langchain4j-core/src/main/java/dev/langchain4j/model/embedding/EmbeddingModel.java]: done

**Pull request merged**